plugins/statistics.koplugin/main.lua

```diff
@@ -328,34 +464,67 @@ function ReaderStatistics:createDB(conn)
                 total_read_time  integer,
                 total_read_pages integer
             );
-        CREATE TABLE IF NOT EXISTS page_stat
-            (
-                id_book    integer,
-                page       integer NOT NULL,
-                start_time integer NOT NULL,
-                period     integer NOT NULL,
-                UNIQUE (page, start_time),
-                FOREIGN KEY(id_book) REFERENCES book(id)
-             );
-        CREATE TABLE IF NOT EXISTS info
-             (
-                 version integer
-             );
-        CREATE INDEX IF NOT EXISTS page_stat_id_book ON page_stat(id_book);
+    ]]
+    conn:exec(sql_stmt)
+
+    -- page_stat_data
+    conn:exec(STATISTICS_DB_PAGE_STAT_DATA_SCHEMA)
+
+    sql_stmt = [[
+        -- Indexes
+        CREATE INDEX IF NOT EXISTS page_stat_data_id_book ON page_stat_data(id_book);
```

Comment:
No more necessary now that we have `UNIQUE (id_book, page, start_time),` (which creates implicitely an index - more precise but as usable as this one).
@marek-g : do you agree?

Reply:
That was indeed going to be my next question ^^

Reply:
I think you are right, but it would be good to verify it with execution plan to ensure that SQLite will use the unique index.

Reply:
Let me test it.

Reply:
Yes. The `page_stat_data_id_book` is not needed. SQLite will happily use the UNIQUE index when quering by id_book.